packages/loot-core/src/server/accounts/sync.ts

```diff
@@ -759,29 +759,44 @@ export async function syncGoCardlessAccount(
       'SELECT date FROM v_transactions WHERE account = ? ORDER BY date ASC LIMIT 1',
       [id],
     );
-    const startingDate = db.fromDateRepr(startingTransaction.date);
+    const startingDate = monthUtils.parseDate(
+      db.fromDateRepr(startingTransaction.date),
+    );
     // assert(startingTransaction)
 
-    // Get all transactions since the latest transaction, plus any 5
-    // days before the latest transaction. This gives us a chance to
-    // resolve any transactions that were entered manually.
-    //
-    // TODO: What this really should do is query the last imported_id
-    // and since then
-    let date = monthUtils.subDays(db.fromDateRepr(latestTransaction.date), 31);
-
-    // Never download transactions before the starting date. This was
-    // when the account was added to the system.
-    if (date < startingDate) {
-      date = startingDate;
-    }
+    const lastImportedTransaction = await db.first(
+      'SELECT date FROM v_transactions WHERE account = ? AND imported_id IS NOT NULL ORDER BY date DESC LIMIT 1',
+      [id],
+    );
+    const lastImportedDate = lastImportedTransaction?.date
+      ? db.fromDateRepr(lastImportedTransaction.date)
+      : null;
+
+    const startDate = monthUtils.dayFromDate(
+      dateFns.max([
+        // Many GoCardless integrations do not support getting more than 90 days
+        // worth of data, so make that the earliest possible limit.
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏what if we instead always fetched the last 90 days?

Bear with me here: I know it's not optimal and adds a few more seconds to the request, but it also lets us keep the code super simple. This also would help us solve edge cases not covered by the current logic. One such example I can think of: car rental. When I rent a car the transaction is cleared only after returning the rental. Which in some cases might be 2 weeks later (which is above the 7 day limit we have here).

WDYT?

Reply:
That is a very convincing reason.

Though the further back we import data the more sad I get that the current merge algorithm ends up unconditionally overriding the transaction date.  Some payment processors take multiple days to submit transactions, and then do so without the original transaction date included, and in those cases I prefer having the date I actually used my card, but the date I set ends up being overwritten by the merge algorithm.

Reply:
Though there is one thing that I personally would find unfortunate about always doing 90 days.

Currently if you have transactions we will never import transactions from before the first transaction.  This makes it easy to make a clean start when importing transactions.  If we switch it to unconditionally doing 90 days then you'd be forced to set up the budget for the past 3 months, or wait 3 months.

So I think it would make sense to do 90 days capped by the first transaction.

Reply:
> So I think it would make sense to do 90 days capped by the first transaction.

Good idea. Can we do that?

> Though the further back we import data the more sad I get that..

We could change the algorithm. When doing bank-sync imports I don't think we should be updating any fields (except for cleared state) of the existing transactions. This functionality only makes sense for manual csv/ofx imports.

Reply:
@MatissJanis This PR now fetches up to 90 days limited by the initial transaction date.  #1559 makes it so that we keep the existing transaction date when merging transactions.

Reply:
@kyrias I noticed something different about Revolut in Croatia. It allows fetching 730 days of transactions. I tested Firefly III and its importer and the helpful feature is that it says how many days can be imported per bank via GoCardless and sets the default value (90) but you can change it during the import/sync. Maybe something similar could be added here?

Reply:
The question is, do we want it to always fetch the maximum allowed, or do we want to give the user an option to select the max time period?  The former would be easy enough, but processing 730 days of transaction on every sync seems rather excessive, whereas the later seems like it would take some potentially non-obvious UI work if we want the user to be able to _temporarily_ raise the limit and then default to something lower.

Anyway, yes, a feature request would be good.

Reply:
Not always but the initial sync would be good to sync max transactions or at least to give a message like "Hey, the maximum days that you can sync with %bank_name% is %max_sync_days%, Would you like to use that or the default - 90?".

Will create an FR to move the discussion there.